File: frontend/src/modules/book/dto/responses/get-book-with-reviews-response.ts
export class GetBookWithReviewsResponse {
  id: number;
  title: string;
  cover: string;
  content: string;
  genre: string;
  author: string;
  rating: number;
  reviews: GetBookWithReviewsResponseReview[];

  constructor(
    id: number,
    title: string,
    cover: string,
    content: string,
    genre: string,
    author: string,
    rating: number,
    reviews: GetBookWithReviewsResponseReview[]) {

    this.id = id;
    this.title = title;
    this.cover = cover;
    this.content = content;
    this.genre = genre;
    this.author = author;
    this.rating = rating;
    this.reviews = reviews;
  }

}

export class GetBookWithReviewsResponseReview {
  id: number;
  message: string;
  reviewer: string;

  constructor(
    id: number,
    message: string,
    reviewer: string) {

    this.id = id;
    this.message = message;
    this.reviewer = reviewer;
  }

}

File: backend/src/LibraryBackend/DTOs/Responses/GetBookWithReviewsResponse.cs
namespace LibraryBackend.DTOs.Responses;

public class GetBookWithReviewsResponse
{
    public class Review
    {
        public int Id { get; init; }

        public string Message { get; init; } = null!;

        public string Reviewer { get; init; } = null!;
    }

    public int Id { get; init; }

    public string Title { get; init; } = null!;

    public string Author { get; init; } = null!;

    public string Cover { get; init; } = null!;

    public string Content { get; init; } = null!;

    public string Genre { get; init; } = null!;

    public double Rating { get; init; }

    public IReadOnlyList<Review> Reviews { get; init; } = null!;
}
